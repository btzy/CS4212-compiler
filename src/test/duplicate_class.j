class Main {

Void main(Int i, Int a, Int b,Int d){

    while(true){

        b = 340 ;

        t1 = t2 ;
    }
}
}


class Dummy {

}

class Dummy {

}
class Dummy {
    Int x;
    String y;
}