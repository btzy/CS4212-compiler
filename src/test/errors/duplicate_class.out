Error: Duplicate declaration of class "Dummy" ( line 19 col 1 - line 19 col 11 ) :
19 | class Dummy {
   | ^^^^^^^^^^^
Note: Previously declared here ( line 15 col 1 - line 15 col 11 ) :
15 | class Dummy {
   | ^^^^^^^^^^^
Error: Duplicate declaration of class "Dummy" ( line 22 col 1 - line 22 col 11 ) :
22 | class Dummy {
   | ^^^^^^^^^^^
Note: Previously declared here ( line 15 col 1 - line 15 col 11 ) :
15 | class Dummy {
   | ^^^^^^^^^^^
Semantic checking aborted due to above errors.
